net-utils/src/lib.rs

```diff
@@ -566,20 +570,33 @@ pub fn bind_two_in_range_with_offset_and_config(
     ip_addr: IpAddr,
     range: PortRange,
     offset: u16,
-    sock1_config: SocketConfig,
-    sock2_config: SocketConfig,
+    mut sock1_config: SocketConfig,
+    mut sock2_config: SocketConfig,
 ) -> io::Result<((u16, UdpSocket), (u16, UdpSocket))> {
     if range.1.saturating_sub(range.0) < offset {
         return Err(io::Error::other(
             "range too small to find two ports with the correct offset".to_string(),
         ));
     }
+    // store original flags
+    let orig_reuseport1 = sock1_config.reuseport;
+    let orig_reuseport2 = sock2_config.reuseport;
+    // clear flags to be able to find actually free ports
+    sock1_config.reuseport = false;
+    sock2_config.reuseport = false;
```

Comment:
I still don't like these config shenanigans. We should respect what gets passed in even if it seems undesirable. If we want to take away some optionality and force the "right" behavior, we should make new APIs and deprecate the broken ones (or fix them and roll the major version)

Reply:
ok I will bump major version then :)

Reply:
the monorepo crates are versioned synchronously. we have a release schedule, versioning cadence and backward compatibility policy. there is zero chance we are bumping major for this

Reply:
Yeah I've figured as much by looking at other PRs. Makes sense to keep everything coherent.